neutrino.go

```diff
@@ -255,6 +255,17 @@ func (sp *ServerPeer) OnVersion(_ *peer.Peer, msg *wire.MsgVersion) *wire.MsgRej
 	// the local clock to keep the network time in sync.
 	sp.server.timeSource.AddTimeSample(sp.Addr(), msg.Timestamp)
 
+	// If the peer doesn't allow us to relay any transactions to them, then
+	// we won't add them as a peer, as they aren't of much use to us.
+	if msg.DisableRelayTx {
+		log.Debugf("%v does not allow transaction relay, disconecting",
+			sp)
+
+		sp.Disconnect()
```

Comment:
Should we also ban them to prevent connecting to them again?

Reply:
IMO no, as this can change for them with just a restart. In the end we aren't fundamentally incompatible. I think we also saw this issue in the past at times when it seemed like things weren't being broadcast, it may have been the case that we were connected to a node running in blocks only mode.

Reply:
Can't the same be said about filter support?

Any reason to not return a `Reject` message with the reason?

Reply:
By just disconnecting without banning, don't we risk going into a connection loop? How likely are we to attempt connecting to this peer again?

Reply:
Seems likely to happen if the address manager continues to return it as part of `newAddressFunc`, which is very aggressive at the moment.